cogs/GuildSettings.py

```diff
@@ -501,141 +512,309 @@ def version_2_patch(all_guild_settings):
             guild_setting.secondary_rating_command = guild_setting.secondary_rating_name
             del guild_setting.__dict__['secondary_rating_name']
-            
+
         guild_setting.type_mapping['primary_rating_command'] = str
         guild_setting.type_mapping['secondary_rating_command'] = str
-            
-        
+
         if 'rating_name' in guild_setting.type_mapping:
             del guild_setting.type_mapping['rating_name']
-            
+
         if 'secondary_rating_name' in guild_setting.type_mapping:
             del guild_setting.type_mapping['secondary_rating_name']
-            
-            
-            
-def version_7_patch(all_guild_settings):        
+
+
+def version_7_patch(all_guild_settings):
     for guild_setting in all_guild_settings.values():
-        
+
         if 'primary_leaderboard_rating_description_text' in guild_setting.__dict__:
             guild_setting.primary_rating_description_text = guild_setting.primary_leaderboard_rating_description_text
-            del guild_setting.__dict__['primary_leaderboard_rating_description_text']
-            
+            del guild_setting.__dict__[
+                'primary_leaderboard_rating_description_text']
+
         if 'secondary_leaderboard_rating_description_text' in guild_setting.__dict__:
             guild_setting.secondary_rating_description_text = guild_setting.secondary_leaderboard_rating_description_text
-            del guild_setting.__dict__['secondary_leaderboard_rating_description_text']
-            
+            del guild_setting.__dict__[
+                'secondary_leaderboard_rating_description_text']
+
         guild_setting.type_mapping['secondary_rating_description_text'] = str
         guild_setting.type_mapping['primary_rating_description_text'] = str
-            
-        
+
         if 'primary_leaderboard_rating_description_text' in guild_setting.type_mapping:
             del guild_setting.type_mapping['primary_leaderboard_rating_description_text']
-            
+
         if 'secondary_leaderboard_rating_description_text' in guild_setting.type_mapping:
             del guild_setting.type_mapping['secondary_leaderboard_rating_description_text']
-            
-            
-
-        
-        
-        
-        
-def save_all_guild_settings():
+
+
+async def save_guild_settings(guild_id):
     global GUILD_SETTINGS
-    pkl_dump_path = "guildsettings_backup.pkl"
-    with open(pkl_dump_path, "wb") as pickle_out:
-        try:
-            p.dump(GUILD_SETTINGS, pickle_out)
-        except:
-            print("Could not dump pickle for guild settings.")
-            
-
-def load_all_guild_settings():
+    guild_settings = GUILD_SETTINGS[guild_id]
+    try:
+        con = await sql.connect('guild_settings.db')
+        await con.execute("PRAGMA foreign_keys = ON")
+        await con.execute(
+            "DELETE FROM guild_settings WHERE guild_id = ?", (guild_id,))

```

Comment:
This works, but in the future, it's cleaner and faster to upsert rather than deleting and inserting.